src/include/compat.h

```diff
@@ -13,10 +13,57 @@
 #define CEPH_COMPAT_H
 
 #if defined(__FreeBSD__)
+
+/* Make sure that ENODATA is defined in the correct way */
+#ifndef ENODATA
 #define	ENODATA	ENOATTR
+#else
+#if (ENODATA == 9919)
+// #warning ENODATA already defined to be 9919, redefining to fix
```

Comment:
@wjwithagen i am not following you.. this does not incur runtime overhead. it's compile-time overhead, iiuc.